lib/storage/metadata/mongoclient/MongoClientInterface.js

```diff
@@ -87,9 +87,17 @@ function generatePHDVersion(versionId) {
 class MongoClientInterface {
     constructor(params) {
         const { replicaSetHosts, writeConcern, replicaSet, readPreference, path,
-            database, logger, replicationGroupId, config } = params;
-        this.mongoUrl = `mongodb://${replicaSetHosts}/?w=${writeConcern}&` +
-            `replicaSet=${replicaSet}&readPreference=${readPreference}`;
+        database, logger, replicationGroupId, authCredentials, config }
+        = params;
+        let cred = '';
+        if (authCredentials &&
+            authCredentials.username &&
+            authCredentials.password) {
+            cred = `${authCredentials.username}:${authCredentials.password}@`;
```

Comment:
Have we validated and encoded previously for the username/password containing a `@` character or other such special characters? Otherwise it sounds like we'd want to encode that here using [percent encoding](https://tools.ietf.org/html/rfc3986#section-2.1).